test/adapter/json_api/type_test.rb

```diff
@@ -0,0 +1,61 @@
+require 'test_helper'
+
+module ActiveModel
+  class Serializer
+    module Adapter
+      class JsonApi
+        class TypeTest < ActiveSupport::TestCase
+          class StringTypeSerializer < ActiveModel::Serializer
+            attributes :name
+            type 'profile'
+          end
+
+          class SymbolTypeSerializer < ActiveModel::Serializer
+            attributes :name
+            type :profile
+          end
+
+          def setup
+            @author = Author.new(id: 1, name: 'Steve K.')
+          end
+
+          def with_jsonapi_resource_type inflection
+            old_inflection = ActiveModelSerializers.config.jsonapi_resource_type
+            ActiveModelSerializers.config.jsonapi_resource_type = inflection
+            yield
+          ensure
+            ActiveModelSerializers.config.jsonapi_resource_type = old_inflection
+          end
```

Comment:
the inflection helper should probably be i the private section at the bottom. I know, pre-existing